subsys/bluetooth/mesh/pb_gatt_srv.c

```diff
@@ -289,7 +291,14 @@ int bt_mesh_pb_gatt_srv_adv_start(void)
 	int64_t timestamp = fast_adv_timestamp;
 	int64_t elapsed_time = k_uptime_delta(&timestamp);
 
-	prov_sd_len = gatt_prov_adv_create(prov_sd);
+        uint8_t device_name[DEVICE_NAME_SIZE];
+        size_t device_name_size = bt_gap_get_device_name(device_name, sizeof(device_name));
+	if (device_name_size < 0) {
+		LOG_DBG("Failed to get name (err %d)", device_name_size);
+		return -ENOMEM;
+	}
+
+	prov_sd_len = gatt_prov_adv_create(prov_sd, device_name, device_name_size);
```

Comment:
This code shows how the old option (`BT_LE_ADV_OPT_USE_NAME` and `BT_LE_ADV_OPT_FORCE_NAME_IN_AD`) that auto-included device name in advs and scan responses was so useful even though it was not intended behavior (though could be converted to). Now users should include the name manually allocating extra space on stack, doing memcpy, etc while host could do this under the hood more optimal. We didn't gain anything from removing that feature, but created more problems for users really.